src/celeritas/optical/model/RayleighModel.hh

```diff
@@ -0,0 +1,59 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/RayleighModel.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "../ImportedModelAdapter.hh"
+#include "../Model.hh"
+
+namespace celeritas
+{
+struct ImportOpticalRayleigh;
+struct ImportOpticalProperty;
+
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Set up and launch the optical Rayleigh scattering model interaction.
+ */
+class RayleighModel : public Model
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using SPConstImported = std::shared_ptr<ImportedModels const>;
+    //!@}
+
+    struct Input
+    {
+        std::vector<ImportOpticalProperty> properties;
+        std::vector<ImportOpticalRayleigh> rayleigh;
```

Comment:
These are redundant with the imported data right? We should also be using the already-processed `optical::MaterialParams` rather than the raw import: so we ought to provide the optical material shared pointer.